### DISCUSSION TITLE
Table Alpine component not loading; Alpine Expression Error: isRecordSelected is not defined

### DISCUSSION THREAD

**Discussion by bricehartmann** [filamentphp/filament#12845] [Help]
### Package

Panel builder

### Package Version

v3.2.80

### How can we help you?

I am attempting to use the table builder included with the panel builder package. In the browser javascript console, I am seeing the following error:

```
Uncaught ReferenceError: isRecordSelected is not defined
```

In the network tab of my browser's dev tools, I do not see a request to `public/js/filament/tables/components/table.js`.
I would expect a network request to the above file as part of this line:
```
ax-load-src="{{ \Filament\Support\Facades\FilamentAsset::getAlpineComponentSrc('table', 'filament/tables') }}"
```
in `vendor/filament/tables/resources/views/index.blade.php`.

I found a similar discussion post [here](https://github.com/filamentphp/filament/discussions/9806), but the accepted answer did not solve my problem (doesn't apply because I already have a root div around `{{ $this->table }}` in my view).

I have `@filamentScripts` in my layout file and do not have `@livewireScripts` specified, because my livewire configuration is set to inject assets automatically.

Any help in debugging would be greatly appreciated. Thank you!

**Comment by bricehartmann** [comment]
I have determined that removing the `#[Lazy]` attribute from my Livewire component that renders the table resolves the issue.

I still don't understand why lazy loading of the component would cause this, however.

**Comment by KenKodz** [comment]
I ran into this issue using the default components. It has something to do with deferred loading. I added `->deferLoading(false)` to my `$table` and the issue went away.

**Comment by szegediszilard** [comment]
I ran into the same issue with a filament table widget. I tried adding the ->deferLoading(false) to my table but unfortunately did not solve the issue

**Reply by leandrocfe** [reply]
`php artisan about --only=Filament`
What is the Filament version?

**Reply by laze** [reply]
3.2.130 - the solution was not working for me either.

I downgraded to 3.2.127, and now it is working.

**Reply by tresero** [reply]
Same with me. Something is wrong with 3.2.130

**Reply by ngunyimacharia** [reply]
> 3.2.130 - the solution was not working for me either.
> 
> I downgraded to 3.2.127, and now it is working.

Downgrading to 3.2.127 fixed the issue for me as well. Thank you @laze.

**Comment by peter-mw** [comment]
Same issue 

`ReferenceError: selectedRecords is not defined  at [Alpine] window.pluralize`

**Reply by leandrocfe** [reply]
run `composer update`

**Reply by EG-Mohamed** [reply]
Same after updating to latest version.

**Reply by heikokrebs** [reply]
Any hints what is going on? Could you solve this problem?

**Reply by mikailhamzat** [reply]
> run `composer update`

This worked for me thanks

**Reply by anindyoy** [reply]
i'm facing the same issue, but run composer update didn't solve the problem..
can someone help me

**Reply by dave-laye** [reply]
`composer update` work for me thanks!

**Reply by khumam** [reply]
> `composer update` work for me thanks!

what version do you use? I still got the same problem even I did composer update

**Comment by ameerhamzanawaz** [comment]
Facing the same issue on my local (apache server) it's working fine but on nginx (production server) it's not working


```javascript
module.esm.js:435 
 Alpine Expression Error: table is not defined

Expression: "table"

```

**Comment by Jmaaswinkel00** [comment]
In my case i had to run the filament upgrade command:
`php artisan filament:upgrade`

The issue in my case was because i had old javascript code. which gets updated by running this command in your terminal. Hope this helps some people

**Reply by jafar-albadarneh** [reply]
I had also to run the following after the `filament:upgrade`:
`php artisan view:clear` and `php artisan optimize:clear`

**Comment by manhofmann** [comment]
seems to be related to #15654

**Comment by karan218** [comment]
I have been same issue on filament V2 suddenly alpine code not working on production after deployment it's working on local and other dev env.. but not work on production.. not opening any modal or any filter or anything... please help it's very urgent

**Comment by ameerhamzanawaz** [comment]
It's difficult to find the exact issue, just make sure to do not do any laravel cache, like view:cache, filament cache or anything else

**Comment by marcusr21** [comment]
Still experiencing this issue. Tried everything listed as well as `->spa()` (#15654) within `AdminPanelProvider`. My problem seems to disappear after refreshing the page, however, on first load for the user it isn't a good user experience. 

Happening on both local environment and Laravel Cloud. Has anyone found a permanent solution or a recommended version to lock to?

**Comment by gotzsys** [comment]
Same thing here, I've tried everything already. I'm using an SPA, and the error only occurs on the first load.

**Reply by briancalma** [reply]
I believe this is related to livewire `wire:navigate` error

**Reply by SilvioRamirez** [reply]
> I believe this is related to livewire `wire:navigate` error

This maybe related, I'm having the same issue. I'm using Laravel 12 with Breeze, Livewire, WireUI, and Tenancy for Laravel, I'm using wire:navigate like a SPA, Interestingly, the error disappears after reloading the page. However, it always occurs the first time you visit the page in a fresh browser session.

When I load a page that uses WireUI components for the first time, I get errors like:

Alpine Expression Error: wireui_modal is not defined  
Alpine Expression Error: show is not defined

In my app.js file, I don’t initialize Alpine because when I do, I get an error saying there are two instances of Alpine running.

Currently, in my app.blade.php, this is what I have:

<head>
    ...
    @wireUiScripts
    @vite(['resources/css/app.css', 'resources/js/app.js'])
    @livewireStyles
</head>
...
...
@livewireScripts
@stack('scripts')
</body>
